pkg/object/function/spec/fsm.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package spec
+
+import (
+	"fmt"
+)
+
+type (
+	// functionFSM is a finite state machine for managing faas function.
+	FSM struct {
+		currentState State
+	}
+
+	// Event is the event type generated by CLI or FaaSProvider.
+	Event string
+
+	// State is the FaaSFunction's state.
+	State string
```

Comment:
seems there's a mix-up of `State` and `Status`.

Reply:
```yaml
      percent: 100
  status:
    address:
      url: http://demo10.default.svc.cluster.local
    conditions:
    - lastTransitionTime: "2021-06-22T08:06:27Z"
      status: "True"
      type: ConfigurationsReady
    - lastTransitionTime: "2021-06-22T08:06:27Z"
      status: "True"
      type: Ready
    - lastTransitionTime: "2021-06-22T08:06:27Z"
      status: "True"
      type: RoutesReady
    latestCreatedRevisionName: demo10-00006
    latestReadyRevisionName: demo10-00006
    observedGeneration: 6
    traffic:
    - latestRevision: true
```
As the example in Knatving serving's `ksvc` type resource, they use status for the first layer structure and inner dedicated status filed. 
How about changing it to all using `status`?

Reply:
https://softwareengineering.stackexchange.com/questions/219351/state-or-status-when-should-a-variable-name-contain-the-word-state-and-w. 
```
status is used to describe an outcome of an operation (e.g. success/fail).
state is used to describe a stage in a process (e.g. pending/dispatched).
```
This answer seems to be useful here. I think FaaSController uses `status` to wrap the `state` might be acceptable according to this answer. :-)  @localvar

Reply:
Thanks, that's LGTM.
I don't clearly understand the difference between the two words before, my comment is just to confirm their usage is by purpose.